### PR TITLE
Update BST after changing stats to allow Scalemons Mod mashup

### PR DESCRIPTION
**Pull request by urkerab**
Reported [here](https://www.smogon.com/forums/posts/8646471).

### CODE REVIEW

**Comment by urkerab** [comment]
(Ugh, `species.bst` is readonly but `species.baseStats.X` isn't. Fun...)

**Comment by MacChaeger** [comment]
Somewhat related: having random-teams run the ModifySpecies event as it builds teams so scalemons and other similar mods that modify base stats can be taken into account for things like level in hackmons cup

File: config/formats.ts [hunk 1]
```diff
@@ -1370,12 +1374,14 @@ export const Formats: FormatList = [
 			};
 			const tier = this.toID(species.tier) || 'ou';
 			if (!(tier in boosts)) return;
-			const pokemon: Species = this.dex.deepClone(species);
+			const pokemon = this.dex.deepClone(species);
+			pokemon.bst = 0;
 			const boost = boosts[tier];
 			let statName: StatName;
-			for (statName in pokemon.baseStats) {
+			for (statName in pokemon.baseStats as StatsTable) {
```
Comment (KrisXV) [inline config/formats.ts:1381]: why is this necessary?

Comment (urkerab) [reply config/formats.ts:1381]: Because `pokemon` is no longer a `Species` (so that I can change its `bst`), `pokemon.baseStats` is no longer a `StatsTable`, so `StatName` is no longer the key of it, resulting in the error "The left-hand side of a 'for...in' statement must be of type 'string' or 'any'".

**Comment by Zarel** [comment]
How important is caching the BST? Should we just have a `Dex.getBST(species)` function?

**Comment by KrisXV** [comment]
> How important is caching the BST? Should we just have a `Dex.getBST(species)` function?

it seems to be done pretty often throughout the PR, so this might be helpful, though I'm unsure how it'd be approached
